packaging/linux/scripts/after-install-and-upgrade.sh

```diff
@@ -58,9 +58,14 @@ BK_IS_UBUNTU_14_10=$?
 command -v initctl > /dev/null
 BK_UPSTART_EXISTS=$?
 
-# Check if the system is Amazon linux which has an old and broken upstart.
-[ -f /etc/system-release ] && grep -qi "Amazon Linux" /etc/system-release
-BK_IS_AMAZON_LINUX=$?
+# Check if upstart is version 0.6.5 as seen on Amazon linux, RHEL6 & CentOS-6
+BK_UPSTART_TOO_OLD=0
+if [ $BK_UPSTART_EXISTS -eq 0 ]; then
+  BK_UPSTART_VERSION="$(initctl --version | awk 'BEGIN{FS="[ ()]"} NR==1{print $4}')"
+  if [ "$BK_UPSTART_VERSION" = "0.6.5" ]; then
```

Comment:
So we could merge this in, and if it turns out 0.6.5/0.6.6/0.6.7 need it we could update the conditional? And eventually the AWS images won't need the hack…